tests/Unit/Services/CampaignServiceTest.php

```diff
@@ -25,6 +25,13 @@ public function testGetCampaignPostStatusCounts()
                 $signup->posts()->saveMany(factory(Post::class, 'rejected', 3)->make());
             });
 
+        $posts = Post::all();
+
+        foreach ($posts as $post) {
+            $post->campaign_id = 57;
+            $post->save();
+        }
+
```

Comment:
Hm, I'm also a lil' perplexed why this would just break all of the sudden! 🤔

I think the issue is that we don't pass a `campaign_id` to the posts we're creating via `saveMany`, and so we end up letting the model factory take control and [generate these on it's own](https://github.com/DoSomething/rogue/blob/e767fe012d7318c44716e0fda37e6faafa7929db/database/factories/ModelFactory.php#L30-L32). Can you try making this change up above & see if that works?

```php
// Create 10 signups for the campaign.
factory(Signup::class, 10)->create(['campaign_id' => $testCampaign['id']])
    ->each(function (Signup $signup) use ($testCampaign) {
        // And for each signup, make 3 accepted, 3 rejected, and 3 pending posts.
        $signup->posts()->saveMany(factory(Post::class, 3)->make(['campaign_id' => $testCampaign['id']]));
        $signup->posts()->saveMany(factory(Post::class, 'accepted', 3)->make(['campaign_id' => $testCampaign['id']]));
        $signup->posts()->saveMany(factory(Post::class, 'rejected', 3)->make(['campaign_id' => $testCampaign['id']]));
    });
```

Reply:
> I think the issue is that we don't pass a campaign_id to the posts we're creating via saveMany, and so we end up letting the model factory take control and generate these on it's own.

@DFurnes that is what I was thinking too but not sure why it would work before with the old query fine and not with this one? 🤔 

Anyway, made the change that you suggested and I like that way better! Didn't know you could do `->make` - awesome!! Updated and thanks for taking a peek at this!